app/src/main/java/com/github/se/icebreakrr/ui/profile/ProfileEdit.kt

```diff
@@ -126,14 +141,18 @@ fun ProfileEditingScreen(
           Column(
               modifier = Modifier.padding(it).padding(padding).testTag("profileEditScreenContent"),
               horizontalAlignment = Alignment.CenterHorizontally) {
-                AsyncImage(
-                    model = profilePictureUrl,
-                    contentDescription = "Profile Picture",
-                    modifier =
-                        Modifier.size(profilePictureSize)
-                            .clip(CircleShape)
-                            .background(MaterialTheme.colorScheme.primary)
-                            .testTag("profilePicture"))
+                // A composable that allows the user to preview and edit a profile picture
+                ProfilePictureSelector(
+                    url = user.profilePictureUrl,
+                    localBitmap = tempBitmap,
+                    size = profilePictureSize,
+                    onSelectionSuccess = { uri ->
+                      profilesViewModel.generateTempProfilePictureBitmap(context, uri)
+                      isModified = true
+                    },
+                    onSelectionFailure = {
+                      Toast.makeText(context, "Failed to select image", Toast.LENGTH_SHORT).show()
+                    })
```

Comment:
help: @louiscrc this is the last lines I have to test to reach 80% coverage but after 1hours on it I still can't find how to test them in a useful way, could you help me?

Reply:
NVM, the actual line coverage is 80%. I was reading the weird coverage displayed on GitHub which is not really the line coverage